### PR TITLE
EBS functions

### PR DESCRIPTION
**Pull request by root-beer**
## Features
- [x] Creating new volume (from snapshot)
- [x] Detaching volume from ec2 instance
- [x] Choose whether or not to treat the alert message as html

### CODE REVIEW

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 1]
```diff
@@ -652,7 +698,13 @@ module.exports = Vue.extend({
     },
 
     isStartPage: function(){ return (this.page === 0); },
-    isEndPage: function(){ return ((this.page + 1) * this.dispItemSize >= this.rules_summary.length); }
+    isEndPage: function(){ return ((this.page + 1) * this.dispItemSize >= this.rules_summary.length); },
+
+    can_create_volume: function () {
+      return this.volume_options.volume_type
+          && this.volume_options.size
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: __[jshint]__
*  Bad line breaking before '&&'.

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 2]
```diff
@@ -652,7 +698,13 @@ module.exports = Vue.extend({
...
+          && this.volume_options.availability_zone;
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: __[jshint]__
*  Bad line breaking before '&&'.

File: app/models/ec2_instance.rb [hunk 3]
```diff
@@ -161,4 +165,20 @@ def retention_policies
...
+                    size: nil,
+                    iops: nil,
+                    encrypted: nil)
+    client.create_volume(
+      availability_zone: availability_zone,
+      snapshot_id:       snapshot_id,
+      volume_type:       volume_type,
+      size:              size,
+      iops:              iops,
+      encrypted:         encrypted,
```
Comment (skyarch-networks) [inline app/models/ec2_instance.rb]: __[rubocop]__
* Style/TrailingCommaInArguments: Avoid comma after the last parameter of a method call. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://sideci.com/github_repositories/31754199/auto_correct/rubocop/pull_request/295/branch/ebs-function)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by root-beer** [comment]
Please review and merge PR. @Joeper214 🙇

**Comment by Joeper214** [comment]
I'll review now.

**Comment by Joeper214** [comment]
Is there any update process script for this PR? @root-beer

**Comment by Joeper214** [comment]
for example: 
```
$npm i 
$gulp tsd
$gulp ts
` ``

**Comment by root-beer** [comment]
`gulp ts` is needed.

**Comment by Joeper214** [comment]
Maybe we need a selectbox for this option? and getting all the list of availability zones.

av-zone

**Comment by Joeper214** [comment]
Or the creation is intended only for the said infrastructure availability zone?

**Comment by root-beer** [comment]
I think so too.
I'll make it selectbox which have all the list of availability zones.
and default value of it is same as the said ec2 instance of it.

File: spec/controllers/nodes_controller_spec.rb [hunk 4]
```diff
@@ -50,12 +50,16 @@
       status: instance_status,
       block_devices: [],
     }}
+    let(:availability_zones) {{
+      available: []
```
Comment (skyarch-networks) [inline spec/controllers/nodes_controller_spec.rb]: __[rubocop]__
* Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline hash. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://sideci.com/github_repositories/31754199/auto_correct/rubocop/pull_request/295/branch/ebs-function)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by root-beer** [comment]
It's done.
Please review and merge PR. @Joeper214 🙇

**Comment by Joeper214** [comment]
I'll review it on monday.

**Comment by root-beer** [comment]
よろしくお願いします 🙇

**Comment by Joeper214** [comment]
LGTM! Thanks. @root-beer :bow:

**Pull request merged**